ray_julia_jll/test/reference_counting.jl

```diff
@@ -0,0 +1,26 @@
+@testset "Reference counting" begin
+    using ray_julia_jll: FromRandom, ObjectID, Hex, AddLocalReference, RemoveLocalReference,
+                         GetAllReferenceCounts, _keys, _getindex, GetCoreWorker
+
+    worker = GetCoreWorker()
+    # need to convert to hex string because these are just pointers
+    has_count(oid) = Hex(oid) in Hex.(_keys(GetAllReferenceCounts(worker)))
+    local_count(oid) = first(_getindex(GetAllReferenceCounts(worker), oid))
+
+    oid = FromRandom(ObjectID)
+
+    @test !has_count(oid)
+
+    AddLocalReference(worker, oid)
+    @test has_count(oid)
+    @test local_count(oid) == 1
+
+    AddLocalReference(worker, oid)
+    @test local_count(oid) == 2
+
+    RemoveLocalReference(worker, oid)
+    @test local_count(oid) == 1
+
+    RemoveLocalReference(worker, oid)
+    @test !has_count(oid)
```

Comment:
What happens here with `local_count`?

Reply:
and what happens if we try to remove it again?

Reply:
the returned dict will not have the corresponding key, so `local_count` gives zero (because it uses `get`)

Reply:
when I was playing around locally, you can remove refs as many times as you want (the count just never goes below zero)

Reply:
Seems like we should test for removing it again:
```suggestion
    @test !has_count(oid)
    @test local_count(oid) == 0
    
    RemoveLocalReference(worker, oid)
    @test !has_count(oid)
    @test local_count(oid) == 0    
```

Reply:
Is there any reason to have both `has_count` and `local_count`?